src/FastFeatureVector/FeatureVector.scala

```diff
@@ -194,21 +199,39 @@ case class FeatureVector(labelset : Array[String],
         return f
     } */
     def read(iterator: Iterator[String]) {
-        val regex = ("""(.*?)(\+L=("""+labelset.mkString("|")+"""))?[ \t]([^ \t]*)""").r  // .*? is non-greedy
-        // (feature, _, label, value)
-        // matches featurename+L=label 1.0
-        fmap.clear()
-        for (line <- iterator) {
-            val regex(feature, _, label, value) = line
-            if (!fmap.contains(feature)) {
-                fmap(feature) = ValuesMap(0.0, Map())
-            }
-            if (label == null) {
-                fmap(feature).unconjoined = value.toDouble
-            } else {
-                fmap(feature).conjoined(labelToIndex(label)) = value.toDouble // TODO: catch invalid label errors and print labels
-            }
+      val newRegex = ( """(.*?)(\+L=(\:[a-zA-Z0-9-]+))?[ \t]([^ \t]*)""" ).r // .*? is non-greedy
```

Comment:
all the whitespace changes makes this hard to review. I think what's happening here is you're allowing for unseen labels, catching errors here instead of lower down, and that's it? If so, looks good to me.